File: app/src/main/java/co/smartreceipts/android/model/impl/columns/receipts/ReceiptCurrencyCodeColumn.java
package co.smartreceipts.android.model.impl.columns.receipts;

import android.support.annotation.NonNull;

import java.util.List;

import co.smartreceipts.android.currency.PriceCurrency;
import co.smartreceipts.android.model.Receipt;
import co.smartreceipts.android.model.factory.PriceBuilderFactory;
import co.smartreceipts.android.model.impl.columns.AbstractColumnImpl;
import co.smartreceipts.android.sync.model.SyncState;

/**
 * Provides a column that returns the category code for a particular receipt
 */
public final class ReceiptCurrencyCodeColumn extends AbstractColumnImpl<Receipt> {

    public ReceiptCurrencyCodeColumn(int id, @NonNull String name, @NonNull SyncState syncState, long customOrderId) {
        super(id, name, syncState, customOrderId);
    }

    @Override
    public String getValue(@NonNull Receipt receipt) {
        return receipt.getPrice().getCurrencyCode();
    }

    @Override
    @NonNull
    public String getFooter(@NonNull List<Receipt> receipts) {
        final PriceCurrency tripCurrency = receipts.get(0).getTrip().getTripCurrency();
        return new PriceBuilderFactory().setPriceables(receipts, tripCurrency).build().getCurrencyCode();
    }
}

File: app/src/main/java/co/smartreceipts/android/model/impl/columns/receipts/ReceiptCurrencyCodeColumn.kt
package co.smartreceipts.android.model.impl.columns.receipts

import co.smartreceipts.android.model.Receipt
import co.smartreceipts.android.model.factory.PriceBuilderFactory
import co.smartreceipts.android.model.impl.columns.AbstractColumnImpl
import co.smartreceipts.android.sync.model.SyncState
import java.util.*

/**
 * Provides a column that returns the category code for a particular receipt
 */
class ReceiptCurrencyCodeColumn(id: Int, syncState: SyncState, customOrderId: Long, uuid: UUID) :
    AbstractColumnImpl<Receipt>(
        id,
        ReceiptColumnDefinitions.ActualDefinition.CURRENCY,
        syncState,
        customOrderId,
        uuid
    ) {

    override fun getValue(rowItem: Receipt): String = rowItem.price.currencyCode

    override fun getFooter(rows: List<Receipt>): String {
        return if (rows.isNotEmpty()) {
            val tripCurrency = rows[0].trip.tripCurrency
            PriceBuilderFactory().setPriceables(rows, tripCurrency).build().currencyCode
        } else {
            ""
        }
    }
}
